File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram (std::string s, std::string t) {
  std::unordered_map <char, int> m; 

  if (s.size() != t.size())  {
    std::cout << "false";
    return false; 
  }
    
  for (int i = 0; i < s.size(); i++) {
    m[s[i]]++; 
  }

  for (int i = 0; i < t.size(); i++) {
    m[t[i]]--; 
  }

  for (auto i : m) {
    if (i.second != 0) {

      std::cout << "false"; 
      return false; 
    }
  }
  
  std::cout << "true"; 
  return true; 
}
};

File: 0242-valid-anagram/0242-valid-anagram.swift
class Solution {
    func isAnagram(_ s: String, _ t: String) -> Bool {
        if s.count != t.count {
            return false
        }

        var sMap: [Character: Int] = [:]
        var tMap: [Character: Int] = [:]
        
        for char in s {
            sMap[char, default: 0] += 1
            tMap[char, default: 0] += 1
        }

        for char in t {
            if let val = tMap[char] {
                tMap[char] = val - 1
            } else {
                return false
            }

            if let val2 = tMap[char] {
                if val2 < 0 {
                    return false
                }
            }
        }

        return true
    }
}